lib/web/integrations_awsoidc.go

```diff
@@ -109,11 +109,33 @@ func (h *Handler) awsOIDCDeployService(w http.ResponseWriter, r *http.Request, p
 		return nil, trace.Wrap(err)
 	}
 
-	databaseAgentMatcherLabels := make(types.Labels, len(req.DatabaseAgentMatcherLabels))
+	databaseAgentMatcherLabels := make(types.Labels, len(req.DatabaseAgentMatcherLabels)+3)
 	for _, label := range req.DatabaseAgentMatcherLabels {
 		databaseAgentMatcherLabels[label.Name] = utils.Strings{label.Value}
 	}
 
+	// DELETE in 19.0: delete only the outer if block (checking labels == 0).
+	// The outer block is required since older UI's will not
+	// send these values to the backend, but instead send custom labels (the UI
+	// will require at least one label before proceeding).
+	// Newer UI's will not send any labels, but instead send the required
+	// fields for default labels.
+	if len(req.DatabaseAgentMatcherLabels) == 0 {
+		if len(req.VPCID) == 0 {
+			return nil, trace.BadParameter("vpc ID is required")
+		}
+		if len(req.Region) == 0 {
+			return nil, trace.BadParameter("AWS region is required")
+		}
+		if len(req.AccountID) == 0 {
+			return nil, trace.BadParameter("AWS account ID is required")
+		}
```

Comment:
💅 
Checking for empty strings is usually done with ` == ""` instead of using `len`.
I had to double check if those variables were string or lists 😅